tests/test_fields_currency.py

```diff
@@ -311,3 +314,102 @@ def test_operator_less_than_or_equal_to(self):  # type: () -> None
             error.pointer,
             'value',
         )
+
+
+class TestAmountDictionariesAndStrings(object):
+    def test_amount_response_dict(self):
```

Comment:
nit: Shall we test the `introspect` method here as well? Just like `test_amount_string` does.

Reply:
`introspect` is already tested by way of this extending `Dictionary`, whose `introspect` is tested in another test.

Reply:
Ah, I see. That makes sense.